modules/openxr/scene/openxr_hand.cpp

```diff
@@ -160,30 +188,102 @@ void OpenXRHand::_get_bones() {
 		"Little_Tip",
 	};
 
+	// The Godot Humanoid skeleton rig bone names ("Left" or "Right" will be prepended).
+	static const char *humanoid_bone_names[XR_HAND_JOINT_COUNT_EXT] = {
+		"Palm",
+		"Hand",
+		"ThumbMetacarpal",
+		"ThumbProximal",
+		"ThumbDistal",
+		"ThumbTip",
+		"IndexMetacarpal",
+		"IndexProximal",
+		"IndexIntermediate",
+		"IndexDistal",
+		"IndexTip",
+		"MiddleMetacarpal",
+		"MiddleProximal",
+		"MiddleIntermediate",
+		"MiddleDistal",
+		"MiddleTip",
+		"RingMetacarpal",
+		"RingProximal",
+		"RingIntermediate",
+		"RingDistal",
+		"RingTip",
+		"LittleMetacarpal",
+		"LittleProximal",
+		"LittleIntermediate",
+		"LittleDistal",
+		"LittleTip",
+	};
+
 	// reset JIC
 	for (int i = 0; i < XR_HAND_JOINT_COUNT_EXT; i++) {
-		bones[i] = -1;
+		joints[i].bone = -1;
+		joints[i].parent_joint = -1;
 	}
 
 	Skeleton3D *skeleton = get_skeleton();
 	if (!skeleton) {
 		return;
 	}
 
-	// We cast to spatials which should allow us to use any subclass of that.
+	// Find the skeleton-bones associated with each OpenXR joint.
+	int bones[XR_HAND_JOINT_COUNT_EXT];
 	for (int i = 0; i < XR_HAND_JOINT_COUNT_EXT; i++) {
-		String bone_name = bone_names[i];
-		if (hand == 0) {
-			bone_name += String("_L");
+		// Construct the expected bone name.
```

Comment:
I guess thought that it'll be a bit harder to make this more generic, unless we add both left and right hand names to the arrays.. Or I guess we could use `String::format` and do something like:
```
const String formats[SKELETON_RIG_MAX][2] = {
  { "%s_L", "%s_R" },
  { "Left%s", "Right%s" }
}
```

Reply:
Nice idea - I modified it slightly to:
```cpp
static const String bone_name_formats[SKELETON_RIG_MAX][2] = {
    { "<bone>_L", "<bone>_R" },
    { "Left<bone>", "Right<bone>" }
};
```

Using "<bone>" makes it a little clearer what's going in, but also lets me  use String::replace which doesn't need the construction of an Array variant for parameter-passing

Reply:
I think that's a great way to do it! Indeed makes it a little easier to read too.